atom/browser/atom_resource_dispatcher_host_delegate.cc

```diff
@@ -124,7 +125,10 @@ bool AtomResourceDispatcherHostDelegate::ShouldInterceptResourceAsStream(
     std::string* payload) {
   const content::ResourceRequestInfo* info =
       content::ResourceRequestInfo::ForRequest(request);
-  if (mime_type == "application/pdf" && info->IsMainFrame()) {
+  content::WebContents* web_contents =
+      info->GetWebContentsGetterForRequest().Run();
+  if (mime_type == "application/pdf" && info->IsMainFrame() &&
+      WebContentsPreferences::IsPluginsEnabled(web_contents)) {
```

Comment:
`WebContentsPreferences` is not thread safe, the check should be moved to `OnPdfResourceIntercepted` and implement a `webContents::DownloadURL` like call when plugin is disabled so that `will-download` event of session is triggered.

```
if (!WebContentsPreferences::IsPluginsEnabled(web_contents)) {
    auto browser_context = web_contents->GetBrowserContext();
    auto download_manager =
      content::BrowserContext::GetDownloadManager(browser_context);

    download_manager->DownloadUrl(
        content::DownloadUrlParameters::CreateForWebContentsMainFrame(
            web_contents, original_url));
    return;
  }
```